sublime_jedi/utils.py

```diff
@@ -214,3 +215,18 @@ def unique(items, pred=lambda x: x):
             continue
         stack.add(calculated)
         yield i
+
+
+def debounce(wait):
+    def decorator(fn):
+        def debounced(*args, **kwargs):
+            def call():
+                fn(*args, **kwargs)
```

Comment:
I worry that it does not return a result

Reply:
Yeah, I agree, it's a little weird. I stole the [implementation from this gist](https://gist.github.com/walkermatt/2871026).

It uses a thread so it can wait before invoking the debounced function without blocking the main thread.

This means it's not really possible to return a value from this function. We would need an event loop like asyncio.

Fortunately, the only method we're wrapping with debounce, `highlight_usages`, returns `None`.

Reply:
👍 
I just wondered if it possible to `debounce` automcompletion also :)

Reply:
That's an interesting question... If we passed a callback into the debounced function my guess it that we could invoke the callback after waiting. this would allow us to debounce calls to autocomplete, which would definitely be a good thing to try.

In the same gist, there are also implementations that apparently don't use threads, like this one: https://gist.github.com/esromneb/8eac6bf5bdfef58304cb.

I'll experiment with it later and see how it works.